library/pk.c

```diff
@@ -460,6 +477,13 @@ int mbedtls_pk_verify_restartable(mbedtls_pk_context *ctx,
         return MBEDTLS_ERR_PK_TYPE_MISMATCH;
     }
 
+
+#if defined(MBEDTLS_USE_PSA_CRYPTO)
+    if (ctx->pk_info->type != MBEDTLS_PK_RSA) {
```

Comment:
Hmm, is this really the right condition? What about `PK_RSA_ALT`, `PK_OPAQUE`, new types we might add in the future? Shouldn't we rather test for equality with one of the types we know works?